src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -831,21 +834,18 @@ private static int compareImports(String import1, String import2) {
      *        One-based line number of previous import.
      * @param toLineNo
      *        One-based line number of current import.
-     * @return count of empty lines between given parameters.
+     * @return count of empty lines between given parameters, exclusive,
+     *        eg., (fromLineNo, toLineNo).
      */
     private int getCountOfEmptyLinesBetween(int fromLineNo, int toLineNo) {
         int result = 0;
         final String[] lines = getLines();
 
-        //  "- 2" because the numbering starts from zero
-        for (int i = toLineNo - 2; i > fromLineNo - 2; i--) {
+        // (fromLineNo - 1, toLineNo - 1 - 1]
```

Comment:
For me, this code fragment is still unobvious. Here is a loop starting from `fromLineNo + 1` and ending to `toLineNo -1`. But due to the fact that lines are zero-based, the index should be decreased by `1`. So it basically
```
        for (int i = fromLineNo + 1; i <= toLineNo - 1; i++) {
            if (CommonUtil.isBlank(lines[i - 1])) {
            }
        }
```
but with optimization. Perhaps there is no need for optimization. The gain is doubtful, but the readability of the code has worsen greatly.

Reply:
we value readability more than nano-optimization.
please update code.
please regenerate diff-report to make sure that refactoring does not change logic.

Reply:
I'm confident about the refactoring 😄

Reply:
Please regenerate report, we had case in our history when very minor refactoring damaged functionality.

Reply:
I'm running the generating yet, it requires about ten minutes :)

Reply:
@romani 
Regenerated the regression report, see https://gmshake.github.io/checkstyle_contribution/checkstyle-tester/diff_pr_7365_4/

Reply:
@pbludov @romani CI looks good :)